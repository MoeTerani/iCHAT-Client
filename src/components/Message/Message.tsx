import React from 'react';
//@ts-expect-error
import ReactEmoji from 'react-emoji';
import './Message.css';

interface Props {
  message: any;
  name: string;
}

const Message = ({ message, name }: Props) => {
  let isSentByCurrentUser = false;

  const trimedUserName = String(name).trim().toLowerCase();

  if (message.user === trimedUserName) {
    isSentByCurrentUser = true;
  }
  return isSentByCurrentUser ? (
    <div className='messageContainer justifyEnd'>
      <p className='sentText pr-10'>{trimedUserName}</p>
      <div className='messageBox backgroundBlue'>
        <p className='messageText colorWhite'>
          {ReactEmoji.emojify(message.text)}
        </p>
      </div>
    </div>
  ) : (
    <div className='messageContainer justifyStart'>
      <div className='messageBox backgroundLight'>
        <p className='messageText colorDark'>
          {ReactEmoji.emojify(message.text)}
        </p>
      </div>
      <p className='sentText pl-10 '>{message.user}</p>
    </div>
  );
};

export default Message;
