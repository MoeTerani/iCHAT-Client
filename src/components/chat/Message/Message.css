.messageBox {
  background: #f3f3f3;
  padding: 7px 15px;
  color: white;
  /* display: inline-block; */
  max-width: 80%;
  margin: 10px 0;
}
.left {
  border-radius: 15px 15px 15px 0;
}
.right {
  border-radius: 15px 15px 0 15px;
}

.messageText {
  width: 100%;
  letter-spacing: 0;
  float: left;
  font-size: 1.1em;
  word-wrap: break-word;
}

.messageText img {
  vertical-align: middle;
}

.messageContainer {
  display: flex;
  justify-content: flex-end;
  padding: 0 5%;
  margin-top: 3px;
}

.sentText {
  display: flex;
  align-items: center;
  font-family: Helvetica;
  color: #828282;
  letter-spacing: 0.3px;
  font-size: 12px;
}

.pl-10 {
  padding-left: 10px;
}

.pr-10 {
  padding-right: 10px;
}

.justifyStart {
  justify-content: flex-start;
}

.justifyEnd {
  justify-content: flex-end;
}

.colorWhite {
  color: white;
}

.colorDark {
  color: #353535;
}

.backgroundBlue {
  background: #2979ff;
}

.backgroundLight {
  background: #f3f3f3;
}

/*--------------------
Messages
--------------------*/
.messages {
  -webkit-box-flex: 1;
  flex: 1 1 auto;
  color: rgba(255, 255, 255, 0.5);
  overflow: hidden;
  position: relative;
  width: 100%;
}
.messages .messages-content {
  position: absolute;
  top: 0;
  left: 0;
  height: 101%;
  width: 100%;
}
.messages .message {
  clear: both;
  float: left;
  padding: 6px 10px 7px;
  border-radius: 10px 10px 10px 0;
  background: rgba(0, 0, 0, 0.3);
  margin: 8px 0;
  font-size: 11px;
  line-height: 1.4;
  margin-left: 35px;
  position: relative;
  text-shadow: 0 1px 1px rgba(0, 0, 0, 0.2);
}
.messages .message .timestamp {
  position: absolute;
  bottom: -15px;
  font-size: 9px;
  color: rgba(255, 255, 255, 0.3);
}
.messages .message::before {
  content: '';
  position: absolute;
  bottom: -6px;
  border-top: 6px solid rgba(0, 0, 0, 0.3);
  left: 0;
  border-right: 7px solid transparent;
}
.messages .message .avatar {
  position: absolute;
  z-index: 1;
  bottom: -15px;
  left: -35px;
  border-radius: 30px;
  width: 30px;
  height: 30px;
  overflow: hidden;
  margin: 0;
  padding: 0;
  border: 2px solid rgba(255, 255, 255, 0.24);
}
.messages .message .avatar img {
  width: 100%;
  height: auto;
}
.messages .message.message-personal {
  float: right;
  color: #fff;
  text-align: right;
  background: linear-gradient(120deg, #248a52, #257287);
  border-radius: 10px 10px 0 10px;
}
.messages .message.message-personal::before {
  left: auto;
  right: 0;
  border-right: none;
  border-left: 5px solid transparent;
  border-top: 4px solid #257287;
  bottom: -4px;
}
.messages .message:last-child {
  margin-bottom: 30px;
}
.messages .message.new {
  -webkit-transform: scale(0);
  transform: scale(0);
  -webkit-transform-origin: 0 0;
  transform-origin: 0 0;
  -webkit-animation: bounce 500ms linear both;
  animation: bounce 500ms linear both;
}
.messages .message.loading::before {
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%, -50%);
  transform: translate(-50%, -50%);
  content: '';
  display: block;
  width: 3px;
  height: 3px;
  border-radius: 50%;
  background: rgba(255, 255, 255, 0.5);
  z-index: 2;
  margin-top: 4px;
  -webkit-animation: ball 0.45s cubic-bezier(0, 0, 0.15, 1) alternate infinite;
  animation: ball 0.45s cubic-bezier(0, 0, 0.15, 1) alternate infinite;
  border: none;
  -webkit-animation-delay: 0.15s;
  animation-delay: 0.15s;
}
.messages .message.loading span {
  display: block;
  font-size: 0;
  width: 20px;
  height: 10px;
  position: relative;
}
.messages .message.loading span::before {
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%, -50%);
  transform: translate(-50%, -50%);
  content: '';
  display: block;
  width: 3px;
  height: 3px;
  border-radius: 50%;
  background: rgba(255, 255, 255, 0.5);
  z-index: 2;
  margin-top: 4px;
  -webkit-animation: ball 0.45s cubic-bezier(0, 0, 0.15, 1) alternate infinite;
  animation: ball 0.45s cubic-bezier(0, 0, 0.15, 1) alternate infinite;
  margin-left: -7px;
}
.messages .message.loading span::after {
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%, -50%);
  transform: translate(-50%, -50%);
  content: '';
  display: block;
  width: 3px;
  height: 3px;
  border-radius: 50%;
  background: rgba(255, 255, 255, 0.5);
  z-index: 2;
  margin-top: 4px;
  -webkit-animation: ball 0.45s cubic-bezier(0, 0, 0.15, 1) alternate infinite;
  animation: ball 0.45s cubic-bezier(0, 0, 0.15, 1) alternate infinite;
  margin-left: 7px;
  -webkit-animation-delay: 0.3s;
  animation-delay: 0.3s;
}

/*--------------------
Avatar
--------------------*/

.message--avatar {
  /* position: absolute;
  z-index: 1;
  top: 8px;
  left: 9px; */
  border-radius: 30px;
  width: 30px;
  height: 30px;
  overflow: hidden;
  margin: 0;
  padding: 0;
  border: 2px solid rgba(255, 255, 255, 0.24);
}
.message--avatar img {
  width: 100%;
  height: auto;
}
