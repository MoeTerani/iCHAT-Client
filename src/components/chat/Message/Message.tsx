import React from 'react';
//@ts-expect-error
import ReactEmoji from 'react-emoji';
import './Message.css';

interface Props {
  message: any;
  name: string;
}

const Message = ({ message, name }: Props) => {
  let isSentByCurrentUser = false;

  const trimedUserName = String(name).trim().toLowerCase();

  if (message.user === trimedUserName) {
    isSentByCurrentUser = true;
  }
  return isSentByCurrentUser ? (
    <div className='messageContainer justifyEnd '>
      <p className='sentText pr-10'>{message.time}</p>

      <div className='messageBox backgroundBlue right'>
        <p className='messageText colorWhite'>
          {ReactEmoji.emojify(message.text)}
        </p>
      </div>
      <div className='message--avatar right--avatar'>
        <img src={message.avatar} alt='user avatar' />
      </div>
    </div>
  ) : (
    <div className='messageContainer justifyStart '>
      <div className='message--avatar left--avatar'>
        <img src={message.avatar} alt='user avatar' />
      </div>
      <div className='messageBox backgroundLight left'>
        <p className='messageText colorDark'>
          {ReactEmoji.emojify(message.text)}
        </p>
      </div>
      <p className='sentText pl-10 '>{message.time}</p>
    </div>
  );
};

export default Message;
