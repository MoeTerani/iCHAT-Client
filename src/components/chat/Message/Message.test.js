import { shallow } from 'enzyme';
import React from 'react';
import Message from './Message';
import {
  getWrapperShallow,
  getWrapperComponent,
} from '../../../utilities/testHelperFn';

describe('Message components', () => {
  let props;
  let Component;

  beforeEach(() => {
    props = {
      message: {},
      name: '',
    };
    Component = getWrapperShallow(Message, { ...props });
  });
  test('it renders without errors', () => {
    expect(Component).toMatchSnapshot();
  });
  test('should render without errors', () => {
    expect(Component.length).toEqual(1);
  });

  //   test('should display messages', () => {
  //     const expected = {
  //       user: 'johnDoe',
  //       text: ':D',
  //       avatar: 'https://api.adorable.io/avatars/285/johnDoe@adorable.png',
  //       time: '8:42 PM',
  //     };

  //     Component.setProps({ message: expected });

  //     const message = getWrapperComponent(Component, 'Message').childAt(0);
  //     console.log({ message });
  //     expect(message.prop('user')).toEqual('johnDoe');
  //     expect(message.childAt(0).text()).toEqual('johnDoe');
  //     expect(message.childAt(2).text()).toEqual(':D');
  //     expect(message.childAt(1).text()).toEqual('10');
  //   });
});
