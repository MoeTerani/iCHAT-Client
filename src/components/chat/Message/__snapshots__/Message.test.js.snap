// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message components it renders without errors 1`] = `ShallowWrapper {}`;
